File: Mercury/led_numbers.m
:- module led_numbers.
:- interface.
:- import_module io.

:- pred main(io::di, io::uo) is det.

:- implementation.
:- import_module list, string, int, array, char.

:- pred get_leds(list.list(character)::in, list.list(character)::in, 
                 int::in, list.list(string)::out ) is det.
	
get_leds(Ldigits, Ldigits_aux, N, Response) :-
	(
		Leds = array([array([" _  ","| | ","|_| "]),
		              array(["  ","| ","| "]),
		              array([" _  "," _| ","|_  "]),
		              array(["_  ","_| ","_| "]),
		              array(["    ","|_| ","  | "]),
		              array([" _  ","|_  "," _| "]),
		              array([" _  ","|_  ","|_| "]),
		              array(["_   "," |  "," |  "]),
		              array([" _  ","|_| ","|_| "]),
		              array([" _  ","|_| "," _| "])]),
		list.length(Ldigits,Len),
		( if Len > 0 then
			Head = det_head(Ldigits),
			Tail = det_tail(Ldigits),
			char.to_int(Head, Head_I:int),
			Head_N:int = Head_I - 48,
			Line = elem(Head_N, Leds),
			Sub_Line = elem(N, Line),
			get_leds(Tail, Ldigits_aux, N, Result),
			Response = [Sub_Line] ++ Result
		  else if N < 2 then
			get_leds(Ldigits_aux, Ldigits_aux, N+1, Result),
			Response = ["\n"]  ++ Result
		  else if N = 2 then
			Response = ["\n"]
		  else
			Response = [] )
	).	
	
main(!IO) :-
	io.write_string("Enter a number: ",!IO),
	io.read_line_as_string(Result, !IO),
	(	if
			Result = ok(String),
			Num = string.strip(String),
			to_char_list(Num,Ldigits),
			get_leds(Ldigits, Ldigits, 0, Response)
		then
			io.write_string(string.join_list("", Response), !IO)
		else
			io.write_string("Not a number...",!IO)
	).

File: Swift-Linux/LEDNumbers.swift
let leds: [Character:String] = [
	"0" : " _  ,| | ,|_| ",
	"1" : "  ,| ,| ",
	"2" : " _  , _| ,|_  ",
	"3" : "_  ,_| ,_| ",
	"4" : "    ,|_| ,  | ",
	"5" : " _  ,|_  , _| ",
	"6" : " _  ,|_  ,|_| ",
	"7" : "_   , |  , |  ",
	"8" : " _  ,|_| ,|_| ",
	"9" : " _  ,|_| , _| "
];

print("Enter a number: ",terminator:"");
let num = readLine(strippingNewline: true);

var line = [String]();
var led = "";

for i in 0...2{
	for character in num!.characters{
		line = String(leds[character]!)!.characters.split(separator: ",").map(String.init);
		print(line[i], terminator:"");
	}
	print("");
}
